File: plaid-adapter-service/src/main/java/com/batu/model/SyncDataModel.java
package com.batu.model;

import com.batu.plaid_adapter_service.entity.Connection;
import com.plaid.client.model.AccountBase;
import com.plaid.client.model.RemovedTransaction;
import com.plaid.client.model.Transaction;
import java.util.List;
import java.util.Set;
import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public class SyncDataModel {

    private Set<AccountBase> accounts;
    private List<Transaction> addedTransactions;
    private List<Transaction> modifiedTransactions;
    private List<RemovedTransaction> removedTransactions;
    private Connection connection;
}

File: plaid-adapter-service/src/main/kotlin/com/batu/plaid_adapter_service/model/SyncDataModel.kt
package com.batu.model

import com.plaid.client.model.AccountBase;
import com.plaid.client.model.RemovedTransaction;
import com.plaid.client.model.Transaction;
import com.batu.plaid_adapter_service.entity.Connection


data class SyncDataModel(
    val accounts : Set<AccountBase> = emptySet(),
    val addedTransactions : List<Transaction> = emptyList(),
    val modifiedTransactions : List<Transaction> = emptyList(),
    val removedTransactions : List<RemovedTransaction> = emptyList(),
    val connection : Connection
)